core/src/main/java/org/infinispan/util/concurrent/BlockingHandlerImpl.java

```diff
@@ -0,0 +1,235 @@
+package org.infinispan.util.concurrent;
+
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.infinispan.commons.executors.BlockingResource;
+import org.infinispan.executors.LimitedExecutor;
+import org.infinispan.factories.KnownComponentNames;
+import org.infinispan.factories.annotations.ComponentName;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.annotations.Start;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+import org.reactivestreams.Publisher;
+
+import io.reactivex.Flowable;
+import io.reactivex.Scheduler;
+import io.reactivex.schedulers.Schedulers;
+
+@Scope(Scopes.GLOBAL)
+public class BlockingHandlerImpl implements BlockingHandler {
+   private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
+   private static final boolean trace = log.isTraceEnabled();
+
+   @Inject @ComponentName(KnownComponentNames.NON_BLOCKING_EXECUTOR)
+   Executor nonBlockingExecutor;
+   // This should eventually be the only reference to blocking executor
+   @SuppressWarnings("checkstyle:RegexpSinglelineJava")
+   @Inject @ComponentName(KnownComponentNames.BLOCKING_EXECUTOR)
+   Executor blockingExecutor;
+
+   Scheduler blockingScheduler;
+   Scheduler nonBlockingScheduler;
```

Comment:
Could be private

Reply:
Yeah, I actually changed this on a dependent branch, hehe.